src/StatusQ/Components/StatusExpandableSettingsItem.qml

```diff
@@ -0,0 +1,150 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+Rectangle {
+    id: root
+
+    property alias icon: icon
+    property alias primaryText: primaryText
+    property alias secondaryText: secondaryText
+    property alias button: button
+    property alias expandableComponent: expandableRegion.sourceComponent
+
+    property bool isExpandable: true
+    property bool hasButton: false
+    property  bool isExpanded: false
+    property bool hasDropDownArrow: false
```

Comment:
I think we can fine-tune a bunch of these:

1. `hasDropDownArrow` is something we ca probably avoid using some heuristics. For example, if the comonent is exapandable and there's no `button` we know we want to render an arrow. No need to introduce a property for this
2. `hasButton` is probably not necessary in the same vein. If there's no `button.text` we probably don't want to render it.
3. Lastly, I personally prefer `expandable` and `expanded` over `isExpandable` and `isExpanded` (also to adhere to QML built-in properties) but this is no blocker, more a personal preference. Your call :)